src/zns.ts

```diff
@@ -154,6 +156,40 @@ export default class Zns extends NamingService {
     return this.structureResolverRecords(await this.records(domain));
   }
 
+  /**
+   * Resolves the ipfs hash stored in domain record
+   * @param domain - domain name
+   * @throws ResolutionError with 
+   *  - NoIPFSConfigurationFound code when records has no fields for ipfs
+   *  - UnregisteredDomain code when no records was found
+   * @returns a promise that resolves in a value stored under ipfs.html.value or undefined. 
+   */
+  async ipfsHash(domain: string): Promise<string> {
+    const record = await this.structureResolverRecords(await this.records(domain));
```

Comment:
Remember the problem of some records having incorrect keys? Ideally we should not call `structureResolverRecords` if we can avoid it because of this. Lets query the apropriate records key from records:

``` typescript
const records = this.records(domain)
if (!records) throw ...;
const record = records['ipfs.html'];
....
```